### ISSUE TITLE
Bazel CI: rules_docker is failing in downstream

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#7465]
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/813#88b43462-54d6-4fa5-9965-e7998faa27bf

```
//container:image_test                                                   FAILED in 3 out of 3 in 56.5s
```

Error message:
```
======================================================================
FAIL: test_data_path (__main__.ImageTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/bazel/.cache/bazel/_bazel_bazel/b35b706ffbcf2263b6c315cc2be685e3/sandbox/linux-sandbox/514/execroot/io_bazel_rules_docker/bazel-out/k8-fastbuild/bin/container/image_test.runfiles/io_bazel_rules_docker/container/image_test.py", line 320, in test_data_path
    self.assertDigest(img, '2b32e6468a11c89ccbd2c386af3a2bfd4a365b2c1f36c0429d93e5ae048eee04')
  File "/home/bazel/.cache/bazel/_bazel_bazel/b35b706ffbcf2263b6c315cc2be685e3/sandbox/linux-sandbox/514/execroot/io_bazel_rules_docker/bazel-out/k8-fastbuild/bin/container/image_test.runfiles/io_bazel_rules_docker/container/image_test.py", line 61, in assertDigest
    self.assertEqual(img.digest(), 'sha256:' + digest)
AssertionError: 'sha256:c1b8e2f5b2a8f9e5c1381728983ceb9272075813965dd0c57dac1c0aed231fa5' != 'sha256:2b32e6468a11c89ccbd2c386af3a2bfd4a365b2c1f36c0429d93e5ae048eee04'
```

Bisecting result points to 45e68f2f4344afa32f91b835888123cda3a77dc6
https://buildkite.com/bazel/culprit-finder/builds/100#fc0925b5-9c1f-4a9b-a18a-f5f4c9860194

@beasleyr-vmw @aiuto @damienmg

**Comment by meteorcloudy** [comment]
Also pinging rules_docker owner @mattmoor @xingao267, as maybe this issue could be fixed from rules_docker.

**Comment by xingao267** [comment]
I'll take a look. It's likely we just need to fix the shas in the tests as the changing in pkg_tar's timestamp can affect the determinism of the containers.

**Comment by xingao267** [comment]
But I think the fix in rules_docker needs to happen after it is released in Bazel right? Otherwise, these tests will fail with current Bazel release.

**Comment by meteorcloudy** [comment]
@xingao267 Thanks for looking into this, that sounds correct. 
But we'll have to wait about one month for that change to be released, I still want to keep the downstream green. Do you think we can temporarily disable `//container:image_test`?

**Comment by nlopezgi** [comment]
Hi @meteorcloudy,
`//container:image_test` is the largest test that we have, we have been chipping away at it (splitting it off into smaller tests) but it still has the bulk of important validations that are necessary to validate any change that comes in, so turning it off is not an option.

It seems to me that this change to the behavior of pkg_tar is a breaking change and should be guarded behind an incompatible flag. This would allow rule owners (likely not just us) to properly prepare for switching our tests that depend on this from using the old vs new behavior.

**Comment by nlopezgi** [comment]
PR https://github.com/bazelbuild/rules_docker/pull/695 is in which should fix this issue iiuc. @meteorcloudy, could you confirm?

**Comment by aiuto** [comment]
The root issue is that the fix for #7276 changes mtime to a non-zero value for ALL .tar files, and not just the ones created by pkg_tar().  My bad for not seeing the danger when I approved it.  I have PR #7482 in flight which puts the behavior change in pkg_tar() only.  rules_docker should then worked unchanged on either old bazel or the new one. So, all should be good.

**Comment by meteorcloudy** [comment]
Rules docker is green again! Thanks, everyone!
